clang/lib/CodeGen/Targets/SPIR.cpp

```diff
@@ -38,9 +38,20 @@ class CommonSPIRABIInfo : public DefaultABIInfo {
 ABIArgInfo CommonSPIRABIInfo::classifyKernelArgumentType(QualType Ty) const {
   Ty = useFirstFieldIfTransparentUnion(Ty);
 
-  if (getContext().getLangOpts().SYCLIsDevice && isAggregateTypeForABI(Ty)) {
+  if (getContext().getLangOpts().SYCLIsDevice) {
```

Comment:
That kind of doesn't look SYCL specific, but rather specific to generation of correct SPIR-V.
I wonder if can do the change in upstream?

Reply:
It is, kind of: `bool` is invalid as a kernel input in opencl, and this is fine here because SYCL controls the arg passing as well (e.g. how to process the bool argument).